src/filter/RuleHelper.js

```diff
@@ -440,6 +440,23 @@ const exports = class {
       }
     }
 
+    if (options.incDimensions) {
+      const dimensionsFilter = this.createDimensionsFilterFromDataSource_(dataSource);
+      if (dimensionsFilter) {
+        if (mainFilter) {
+          mainFilter = olFormatFilter.and.apply(
+            null,
+            [
```

Comment:
That's personal, but it it useful to have so many return to line ?
I find this harder to read